src/App.tsx

```diff
@@ -1,223 +1,227 @@
     // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    if (user) {
+      getTodos(user.id)
+        .then((loadedTodos) => {
+          setTodos(loadedTodos);
+        })
+        .catch(() => {
+          setIsError(true);
+          showError('Something went wrong');
+        });
+    }
   }, []);
 
+  const deleteTodo = async (todoId: number) => {
+    try {
+      const deleteResponse = await deleteTodoById(todoId);
+
+      // eslint-disable-next-line max-len
+      setTodos((currentTodos) => currentTodos.filter((todo) => todo.id !== todoId));
+
+      return deleteResponse;
+    } catch (error) {
+      setIsError(false);
+      showError('Unable to delete a todo');
+
+      return false;
+    }
+  };
+
+  const addTodo = async (newTitle: string) => {
+    if (!user) {
+      return;
+    }
+
+    setIsAdding(true);
+
+    setTempTodo({
+      id: 0,
+      userId: user.id,
+      title: newTitle,
+      completed: false,
+    });
+
+    const newTodo = {
+      userId: user.id,
+      title: newTitle,
+      completed: false,
+    };
+
+    try {
+      const response = await createTodo(newTodo);
+
+      setTodos((prevTodos) => [...prevTodos, response]);
+
+      // eslint-disable-next-line consistent-return
+      return response;
+    } catch (error) {
+      showError('Unable to add a todo');
+
+      // eslint-disable-next-line consistent-return
+      return false;
+    } finally {
+      setTempTodo(null);
+      setIsAdding(false);
+    }
+  };
+
+  const updateTodo = async (
+    todoId: number,
+    fieldsToUpdate: Partial<Pick<Todo, 'title' | 'completed'>>,
+  ) => {
+    setUpdatingTodoIds((prevIds) => {
+      if (!prevIds.includes(todoId)) {
+        return [...prevIds, todoId];
+      }
+
+      return prevIds;
+    });
+
+    try {
+      const response = await updateTodoOnServer(todoId, fieldsToUpdate);
+
+      setTodos((prevTodo) => prevTodo.map((todo) => {
+        if (todo.id === todoId) {
+          return response;
+        }
+
+        return todo;
+      }));
+    } catch (error) {
+      setErrorMessage('Unable to update a tod');
+    } finally {
+      setUpdatingTodoIds((prevTodos) => (
+        prevTodos.filter((prevTodoId) => prevTodoId !== todoId)));
+    }
+  };
+
+  const visibleTodos = useMemo(
+    () => getFilterTodos(todos, filterStatus),
+    [todos, filterStatus],
+  );
+
+  const notCompletedTodosLength = useMemo(
+    () => todos.filter((todo) => !todo.completed).length,
+    [todos],
+  );
+
+  const completedTodosLength = useMemo(
+    () => todos.filter((todo) => todo.completed).length,
+    [todos],
+  );
+
+  const deleteCompletedTodos = useCallback(() => {
+    todos.forEach((todo) => {
+      if (todo.completed) {
+        deleteTodo(todo.id);
+      }
+    });
+  }, [todos]);
+
+  const isAllTodosCompleted = todos.length === completedTodosLength;
+
+  const handelTodosStatus = useCallback(() => {
```

Comment:
A little typo. But I think toggleTodosStatus would be even more informative. 
```suggestion
  const handleTodosStatus = useCallback(() => {
```